### ISSUE TITLE
Hover-state: should not only convey state using colour (WCAG 2.1 1.4.1)

### ISSUE THREAD

**Issue by Camulos** [digdir/designsystem#822]
### Description

Different states for buttons should not only change color.

Suggestion: add border (3px) at the bottom for hover-state on button, also underline on icons and text. Make sure each state of an interactive element is different from each other.

Relates to WCAG 2.1: 1.4.1 Use of color
Color is not used as the only visual means of conveying information, indicating an action, prompting a response, or distinguishing a visual element.

**Comment by mrosvik** [comment]
Lets have a meeting about this :) @Febakke

**Comment by Febakke** [comment]
I have researched this already in releation to the apps team: https://github.com/Altinn/app-frontend-react/issues/1364
And I did it again now haveing been reading trough WCAG discusions on Github and the 1.4.1 Use of Color and I cant see that it is applying to hover on buttons, radios and other controls. I cannot find other sources claiming this either.

Im not opposing to have an underline on buttons if that makes it clearer on hover but I am opposing that this is breaching the WCAG guidelines and I really want to find an answer to this. Im guessing you can make it clearer for me @Camulos

My understanding of this issue has always been that you do not convey information about something by just using color.
> The intent of this Success Criterion is to ensure that all sighted users can access information that is conveyed by color differences, that is, by the use of color where each color has a meaning assigned to it.

A common example of a failure:  "All red form fields are required" or one of the examples of actions from 1.4.1 Use of Color which is indicating that a database has been succesfuly updated only using color. This is examples of trying to convey information to the user only using color. A button or a radio button has a form, text and color indicating that they are interactive it is not when hovering that this meaning is conveyed. If so all mobile and tablet user would have a lot of trouble navigating the net. 

Even the closest technique to what you are saying [G183](https://www.w3.org/WAI/WCAG21/Techniques/general/G183) is saying hover is beyond 1.4.1 Use of Color.

> To meet success criterion 1.4.1: Use of Color a [relative luminance](https://www.w3.org/TR/WCAG21/#dfn-relative-luminance) (lightness) difference of 3:1 or greater with the text around can be used. **This technique goes beyond the success criterion and asks for visual highlights when the user hovers over each link**, such as an underline, a change in font style such as bold or italics, or an increase in font size.

**Comment by mrosvik** [comment]
For `hover`-state på knapper har vi brukt det vanligste interaksjonsmønsteret som er en liten fargeendring i form av at knappen blir ett hakk mørkere eller lysere. Hvis dette er et brudd på WCAG, så bryter både [Nav](https://aksel.nav.no/komponenter/core/button), [Gov UK](https://design-system.service.gov.uk/components/button/), [US Gov](https://designsystem.digital.gov/components/button/), [Entur](https://design.entur.no/komponenter/knapper/button), Material design (google), Apple og Nielsen-Norman-Group dette. 🤔

For `hover`-state på lenker blir streken under tykkere ved hover.

**Comment by mrosvik** [comment]
Her mener jeg vi bør følge de vanligste konvensjonene. Kan [store endringer](https://www.brreg.no/aksjeselskap/) på `hover` oppleves overveldende dersom det er mange interaktive elementer på en side? Kan bruker bli usikker på om noe endret seg av informasjon eller om hen kom borti og trykket på knappen når det er så store visuelle endringer?

**Comment by Febakke** [comment]
Jeg har nå saumfart alt jeg kan finne av dokumentasjon for å finne noe som sier at dette er et brudd på 1.4.1 Use of color og fra min side kan jeg ikke se det. Om ikke du har noen andre kilder @Camulos? 

Her er et forslag på Githuben til W3/WCAG på å lage et krav for WCAG 3 spesifikt for synligere hover states [#899 ](https://github.com/w3c/wcag/issues/899) Jeg tror ikke denne oppgaven hadde ligget åpen i 4 år om det allerede var dekket av 1.4.1. 

Her fra en annen diskusjon på en annen oppgave: 
> In general (for 2.0-2.2) I can't think of any requirements for hover states. As the user controls the mouse, and can (for example) configure a large mouse pointer, the visibility of the hover state is not something WCAG has set a requirement for.

[Link](https://github.com/w3c/wcag/issues/2900)

Jeg er helt med på å diskutere om vi skal ha tydeligere hover visninger, men vi må ha det på riktig grunnlag. Og sånn som jeg ser det er det ikke et brudd på 1.4.1 Use of Color.

**Comment by tvs-brreg** [comment]
[Forsiden til NAV](https://www.nav.no/) er et godt eksempel av en "fysiske" endring på knapper som ikke er intens. En trenger faktisk ikke så mye som de knappene i selve forsiden en gang. Bare den "fysiske" endringen som er å fjerne underlinja på teksten slik som det er i header menyen eller en bakgrunnsfarge + border slik der er på sekundær knappene er nokk.

**Comment by Febakke** [comment]
@tvs-brreg
Som sagt er jeg ikke i mot å diskutere å legge til styling på hover states. Men jeg er uenig i at det er et brudd på 1.4.1 Use of Color. WCAG reglene er store og komplekse og for mange vanskelig å sette seg inn i. Så de stoler på oss som har brukt tid på å sette oss inn i kravene. Om vi sier at dette må endres fordi det er brudd og noen ser oss i kortene på det vil det være veldig uheldig og potensielt svekke tilliten vi har jobbet med å bygge opp. Kanskje litt vel dramatisk, men jeg syns det er viktig å være tydelig på hva som er faktiske brudd og hva som er forbedringer. 😄

**Comment by mrosvik** [comment]
Konklusjon etter møte 26.09.23
(@Febakke , @Camulos, @Thuneer, @mrosvik )

- Vi er enige om at dette ikke er et brudd på WCAG 2.1 1.4.1. 
- Vi må følge med på https://github.com/w3c/wcag/issues/899 
- Høre med andre om dette er et tema som er diskutert?
- Ikke ta for store beslutninger uten innsikt, som går i mot generelle konvensjoner.
- Vi må finne en helhetlig måte å håndtere dette på, i samarbeid med andre.
- Det må gjøres et researcharbeid på om en slik endring gir verdi, og hvilke brukergrupper det er nyttig for. Det må testes på brukere med nedsatt syn som bruker musepeker og ikke tastatur. 
- Inntil vi får tid til å gjøre mer innsikt, følger vi samme mønsteret som Nav når det kommer til hover på `Button`-komponenten
- Når det gjelder lenker går vi for 1px border default og 2px border hover. (Nav fjerner understrekning på lenker ved hover, men det blir da veldig likt som i løpende tekst og en risikerer at ikke alle ser forskjellen) Vi bør ikke fjerne ting ved hover, men heller legge til. Er dette noe som har blitt diskutert i @[Aksel](https://github.com/navikt/aksel)?
